bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ErrorUtils.java

```diff
@@ -133,4 +144,40 @@ public static BError createUnorderedTypesError(Object lhsValue, Object rhsValue)
         throw createError(BallerinaErrorReasons.UNORDERED_TYPES_ERROR, BLangExceptionHelper.getErrorMessage(
                 RuntimeErrors.UNORDERED_TYPES_IN_COMPARISON, lhsValue, rhsValue));
     }
+
+    public static BError createRuntimeError(Module module, RuntimeErrorType errorType, Object... params) {
+        return createRuntimeError(module, errorType, getErrorDetail(getErrorMessage(errorType, params)), null);
+    }
+
+    public static BError createRuntimeError(Module module, RuntimeErrorType errorType,
+                                            BMap<BString, Object> detail, BError cause) {
+        try {
+            return createError(module, errorType.getErrorName(), getModulePrefixedErrorName(module, errorType),
+                    cause, detail);
+        } catch (BError e) {
+            throw ErrorCreator.createError(StringUtils.fromString(e.getMessage()));
```

Comment:
we need a comment here eg:
`// This should never happen unless ErrorCreator itself has a bug. Fall back to java error in that case`
and let's throw BallerinaException in that case so we don't lose the original error.

Reply:
cause should be `e` and additional info (via `addSupressed`) should include `cause`